test/model.querying.test.js

```diff
@@ -2399,6 +2399,17 @@ describe('model: querying:', function() {
       });
     });
 
+    it('removes the __v property if versionKey: false is set (gh-8934)', async function() {
+      const title = 'Wooooot ' + random();
+      await BlogPostB.create({ title });
+      const foundPost = await BlogPostB.find({ title }).lean({ versionKey: false });
+      assert.strictEqual(foundPost.__v, undefined);
```

Comment:
I think we should actually test if the attribute exists and not if the value is undefined.
Like
```suggestion
      assert.ok(!('__v' in foundPost));
```

Also the other two assertions should be modified accordingly